libraries/lib-numeric-formats/formatters/BeatsNumericConverterFormatter.cpp

```diff
@@ -331,25 +335,39 @@ class BeatsNumericConverterFormatterFactory final :
 
 private:
    const int mFracPart;
+   const bool mTimeFormat;
 };
 
-NumericConverterItemRegistrator beatsTime {
-   Registry::Placement { {}, { Registry::OrderingHint::After, L"parsedTime" } },
-   NumericConverterFormatterGroup(
-      "beats", NumericConverterType_TIME,
+Registry::BaseItemPtr BuildBeatsGroup(bool timeFormat)
+{
+   return NumericConverterFormatterGroup(
+      timeFormat ? "beatsTime" : "beatsDuration",
+      timeFormat ? NumericConverterType_TIME : NumericConverterType_DURATION,
       NumericConverterFormatterItem(
          /* i18n-hint: "bar" and "beat" are musical notation elements. */
          "beats", XO("bar:beat"),
-         std::make_unique<BeatsNumericConverterFormatterFactory>(0)),
+         std::make_unique<BeatsNumericConverterFormatterFactory>(0, timeFormat)),
       NumericConverterFormatterItem(
-         /* i18n-hint: "bar" and "beat" are musical notation elements. "tick" corresponds to a 16th note.  */
+         /* i18n-hint: "bar" and "beat" are musical notation elements. "tick"
+            corresponds to a 16th note.  */
          "beats16", XO("bar:beat:tick"),
-         std::make_unique<BeatsNumericConverterFormatterFactory>(16)))
+         std::make_unique<BeatsNumericConverterFormatterFactory>(16, timeFormat)));
+}
+
+NumericConverterItemRegistrator beatsTime {
+   Registry::Placement { {}, { Registry::OrderingHint::After, L"parsedTime" } },
+   BuildBeatsGroup(true)
+};
+
+NumericConverterItemRegistrator beatsDuration {
+   Registry::Placement { {}, { Registry::OrderingHint::After, L"parsedDuration" } },
+   BuildBeatsGroup(false)
 };
 } // namespace
 
 std::unique_ptr<NumericConverterFormatter> CreateBeatsNumericConverterFormatter(
-   const FormatterContext& context, int fracPart /*= 0*/)
+   const FormatterContext& context, int fracPart /*= 0*/,
+   bool timeFormat /*= true*/)
```

Comment:
This function is called only from unit tests.  Is it easy and sensible to add some test cases?